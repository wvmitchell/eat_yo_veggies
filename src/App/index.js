import React, { Component } from 'react';
import AddVeggiesForm from '../AddVeggiesForm'
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      vegetables: []
    }
  }

  addVegetable = (veggie) => {
    this.setState({ vegetables: [...this.state.vegetables, veggie]})
  }

  render() {
    return (
      <div className="App">
        <AddVeggiesForm addVegetable={this.addVegetable} />
      </div>
    );
  }
}

export default App;
