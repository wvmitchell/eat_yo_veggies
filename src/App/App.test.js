import React from 'react';
import { shallow } from 'enzyme'
import App from './index';

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the default state', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.state()).toEqual({ vegetables: [] })
  })
})
