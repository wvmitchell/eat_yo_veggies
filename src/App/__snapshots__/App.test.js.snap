// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <AddVeggiesForm
    addVegetable={[Function]}
  />
</div>
`;
